pkg/jobstore/boltdb/store.go

```diff
@@ -700,10 +699,6 @@ func (b *BoltJobStore) getJobHistory(tx *bolt.Tx, jobID string,
 			return false
 		}
 
-		if query.NodeID != "" && !strings.HasPrefix(event.NodeID, query.NodeID) {
```

Comment:
If we are getting rid of `NodeID` in query, it would be important to also change that as part of the API request [`ListJobHistoryRequest`](https://github.com/bacalhau-project/bacalhau/blob/main/pkg/publicapi/apimodels/job.go#L110).